egg/core/language_analysis.py

```diff
@@ -28,7 +27,11 @@ def entropy_dict(freq_table):
     True
     """
     t = torch.tensor([v for v in freq_table.values()]).float()
-    return torch.distributions.Categorical(probs=t).entropy().item() / np.log(2)
+    if (t < 0.0).any():
+        raise RuntimeError('Encountered negative probabilities')
+
+    t /= t.sum()
+    return -(torch.where(t > 0, t.log(), t) * t).sum().item() / np.log(2)
```

Comment:
Silly Categorical doesn't allow prob=0, clamping them at some small value

Reply:
yeah, but 
```
>>> torch.distributions.Categorical(probs=torch.tensor([0, 1.0])).entropy().item() / np.log(2)
1.7198267136473827e-07
```
which is positive...

Reply:
how about `round(torch.distributions.Categorical(probs=torch.tensor([0.0, 1.0])).entropy().item() / np.log(2), 4)`

Reply:
looks more magic to me than the original version =D